README.md

```diff
@@ -1,27 +1,508 @@
+    form of external state. Sounds fancy, but essentially it means pure
+    functions can't load data from an API or update some global variable (such
+    as `localStorage`).
+
+### Internal State
+
+As we just saw, props can be seen as arguments for pure components. Pure
+components don't rely on any internal state or side-effects: They simply return
+a virtual representation of how they want to "look" like.
+
+There is only one problem: props are always being "passed down" from the
+component's immediate ancestor. So for example, the `Blog` component passes the
+blog post data to the `BlogPost` component. But where does the blog *data*
+actually come from?
+
+```js
+class Blog extends React.Component {
+    render () {
+        return (
+            <div>
+                <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+                <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+            </div>
+        )
+    }
+}
+```
+
+In the above example, we simply hard-coded the blog titles and bodies. Obviously
+this isn't really what we want, otherwise we might as well just render
+everything on the server — not very *reactive* at all.
+
+Instead we typically want to dynamically load the data from some API endpoint,
+for instance `GET /api/posts`. So whenever we render a new blog component, we
+do a XHR call and load the blog posts that we want to render.
+
+In React, the function that gets executed once a component has been "rendered"
+(= created) is called `componentDidMount`. `componentDidMount` is a lifecycle
+method, as such it is **not** available for pure components (such as the
+`BlogPost` component that we defined using an arrow function):
+
+```js
+class Blog extends React.Component {
+    componentDidMount () {
+        fetch('http://localhost:4000/api/posts')
+            .then(response => response.json())
+            .then(posts => {
+                console.log('Yey! I got some posts!', posts)
+            })
+    }
+    render () {
+        return (
+            <div>
+                <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+                <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+            </div>
+        )
+    }
+}
+```
+
+This means that whenever we `mount` this component by rendering it as a child of
+some other part of our application, e.g. using `<Blog />`, React executes the
+`componentDidMount` method that requests data from our API.
+
+Once we receive those blog posts, we `console.log` them. While
+`console.log`ging them is certainly fun, our users won't be too happy about
+having to open the Chrome DevTools in order to read our blog. So instead of just
+printing them to the console, the `Blog` component should somehow "keep track
+of them".
+
+The most intuitive solution would be to "set" them as a property on the
```

Comment:
As mentioned before, this section can be a lot less verbose since students will already know how to set and update state at this point